### ISSUE TITLE
vis glitch: large audio buffer (or low samplerate) leads to low-frequency visualization updates

### ISSUE THREAD

**Issue by Oleksiy-Yakovenko** [DeaDBeeF-Player/deadbeef#1183]
Original [issue 1282](https://code.google.com/p/ddb/issues/detail?id=1282) created by Alexey-Yakovenko on 2015-02-23T14:42:18.000Z:

<b>What steps will reproduce the problem?</b>
<b>Какие шаги приводят к воспроизведению</b>
<b>проблемы?</b>
1. Set output to pulseaudio
2. have spectrum/musical spectrum component in the UI
3. play any track via internal sound card

<b>What is the expected output? What do you see instead?</b>
<b>Какой ожидаемый вывод? Что вы видите</b>
<b>вместо него?</b>
Expected: smooth animation
Actually: animation will stutter, unless pavucontrol is launched/running in the background

<b>What version of the product are you using? On what operating system and CPU</b>
<b>architecture?</b>
<b>Какую версию продукта вы используете? На</b>
<b>какой операционной системе и</b>
<b>архитектуре CPU?</b>
archlinux x64

<b>How did you install the product?</b>
<b>Как вы установили продукт?</b>
deadbeef-git from AUR

<b>Please provide any additional information below.</b>
<b>Пожалуйста, предоставьте любую</b>
<b>дополнительную информацию ниже.</b>
Basically, ddb -&gt; alsa -&gt; pulse works fine, while ddb -&gt; pulse does not. USB/BT DACs seem to not be affected, only the internal ones.

See here for more details: https://github.com/cboxdoerfer/ddb_musical_spectrum/issues/8

**Comment by Alexey-Yakovenko** [comment]
Converting into a feature, since while it looks like a bug, fixing it pretty much requires developing a wholly new visualization backend.

**Comment by Samueru-sama** [comment]
This issue gets fixed if you set env var `PULSE_LATENCY_MSEC=60`

I believe deadbeef needs to set this automatically, otherwise the default latency is too high. 

I noticed that if you open pavucontrol-qt or play a video in your web browser the issue gets fixed in deadbeef as well while those operations are going, likely because they lower the latency.
